components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/SubscriptionManagerImpl.java

```diff
@@ -52,9 +112,13 @@
                     log.error("Device with ID: " + device.getId() + " not found to install the application.");
                 } else {
                     if (log.isDebugEnabled()) {
-                        log.debug("Installing application to : " + device.getId());
+                        log.debug("Prepare application install to : " + device.getId());
+                    }
+                    if (device.getType().equals(ANDROID)) {
```

Comment:
Do we need to have some device specific check here? The app manager implementation should be generic, and not tied to any device type.

Reply:
We will need this check when sending the install operation to devices. 
Because when creating the operation to Android we will have to call *MDMAndroidOperationUtil.createInstallAppOperation* and when creating the operation to iOS we have to call *MDMIOSOperationUtil.createInstallAppOperation*. Therefore I added this logic before hand.